src/index.ts

```diff
@@ -1,36 +1,96 @@
-import "reflect-metadata"
-import { AppDataSource } from "./data-source"
-import { User } from "./entity/User"
+import 'reflect-metadata';
 import { ApolloServer } from '@apollo/server';
 import { startStandaloneServer } from '@apollo/server/standalone';
+import { AppDataSource } from './data-source';
+import { User } from './entity/User';
+import 'bcrypt';
 
-AppDataSource.initialize().then(async () => {
-
-    const user = new User()
-    user.firstName = "3"
-    user.lastName = ""
-    user.age = 20
-
-    await AppDataSource.manager.save(user)
-})
-
+//GraphQl Schema
+//para colocar input (input NomeInput)
 const typeDefs = `#graphql
-    type Query {
+
+  type Query{
     hello: String
+  }
+
+  #input obrigatório !
+  type Mutation{
+    CreateUser(input: CreateUserInput!): User
 }
+  
+  input CreateUserInput{
+    name: String!
+    email: String!
+    password: String!
+    birthDate: String!
+  }
+
+   type User {
+    id: ID
+    name: String
+    email: String
+    password: String
+    birthDate: String
+  }
 `;
+//interface para conectar args com os inputs
+//typescript como default é obrigatório, para ser "opcional" -> ?
+interface CreateUserInput {
+  input: {
+    name: string;
+    email: string;
+    password: string;
+    birthDate: string;
+  };
+}
 
-  const resolvers = {
-    Query: {
-      hello: () => "Hello World",
+const resolvers = {
+  Query: {
+    hello: () => 'Hello World',
+  },
+  Mutation: {
+    CreateUser: async (parent: unknown, args: CreateUserInput) => {
+      //validação do email - busca o banco se já existe um email igual
+      const storageUsers = await AppDataSource.manager.find(User, { where: { email: args.input.email } });
```

Comment:
como você está procurando um user específico, pode utilizar o `findOne`
```suggestion
      const user = await AppDataSource.manager.findOne(User, { where: { email: args.input.email } });
```